app/scripts/models/ThreeViewer.js

```diff
@@ -0,0 +1,306 @@
+                    pos = i * 3;
+                    positions.push(
+                        list.lineStripVertices[pos],
+                        list.lineStripVertices[pos + 1],
+                        list.lineStripVertices[pos + 2]
+                    );
+                }
+
+                index += count;
+                geometry.addAttribute('position', new THREE.BufferAttribute(new Float32Array(positions), 3));
+            }
+
+            geom.add(new THREE.Line(geometry, lineMat, THREE.LinePieces));
+        }
+
+        if (list.pointVertices.length) {
+
+            geometry.addAttribute('position', new THREE.BufferAttribute(list.pointVertices, 3));
+            mesh = new THREE.PointCloud(geometry, partMat);
+
+            geom.add(mesh);
+        }
+
+        if (list.triangleVertices.length) {
+
+            geometry.addAttribute('position', new THREE.BufferAttribute(list.triangleVertices, 3));
+            geometry.addAttribute('normal', new THREE.BufferAttribute(list.triangleNormals, 3));
+            mesh = new THREE.Mesh(geometry, meshMat);
+
+            geom.add(mesh);
+        }
+    };
+
+    function onRequestGeometry(){
+        if(this.pendingRequestsCount === 0)
+            this.app.trigger('show-progress');
+        this.pendingRequestsCount++;
+    };
+
+    function zoomToFit(){
+        this.viewPort.zoomToFit();
+    };
+
+    function dollyOut(){
+        this.viewPort.dollyOut();
+    };
+
+    function dollyIn(){
+        this.viewPort.dollyIn();
+    };
+
+    function exportSTL(e){
+        GeometryExport.toSTL(this.viewPort.scene, e.name);
+    };
+
+    return Backbone.Model.extend({
+
+        initialize: function (args, options) {
+            this.viewPort = new Viewport();
+            this.viewPort.init();
+
+            this.app = args.app;
+            this.idMap = {};
+
+            this.pendingRequestsCount = 0;
+
+            this.listenTo(this.app, 'zoomToFit', zoomToFit.bind(this));
+            this.listenTo(this.app, 'dollyOut', dollyOut.bind(this));
+            this.listenTo(this.app, 'dollyIn', dollyIn.bind(this));
+            this.listenTo(this.app, 'requestGeometry', onRequestGeometry.bind(this));
+            this.listenTo(this.app, 'geometry-data-received:event', updateNodeGeometry.bind(this));
+            this.listenTo(this.app.get('workspaces'), 'geometryUpdated', updateNodeGeometry.bind(this));
+            this.listenTo(this.app.get('workspaces'), 'nodeSelected', nodeSelected.bind(this));
+            this.listenTo(this.app.get('workspaces'), 'nodeVisible', nodeVisible.bind(this));
+            this.listenTo(this.app.get('workspaces'), 'nodeRemove', nodeRemove.bind(this));
+            this.listenTo(this.app.get('workspaces'), 'changeWorkspace', changeWorkspace.bind(this));
+            this.listenTo(this.app.get('workspaces'), 'clearNodeGeometry', clearNodeGeometry.bind(this));
+            this.listenTo(this.app.get('workspaces'), 'exportSTL', exportSTL.bind(this));
```

Comment:
All of these should be method invocations by `App` on `ThreeView`.  It keeps data flow clearer.  `App` owns `ThreeViewer` so it should not pass a reference to itself into `ThreeViewer`. Conceptually we want a DAG shaped reference hierarchy, if that makes sense.  It helps things compose more cleanly.